{
  "id" : 213,
  "expression" : "new FailOnStartCommandProcessor(\"some crazy unexpected exception\")",
  "projectName" : "SeleniumHQ@selenium",
  "commitID" : "c146621f39dbec94114835aa1f6af6a0e1fa4cd0",
  "filePath" : "/java/client/test/com/thoughtworks/selenium/DefaultSeleniumStartErrorHandlingTest.java",
  "occurrences" : 1,
  "isArithmeticExpression" : 0,
  "isGetTypeMethod" : 0,
  "expressionList" : [ {
    "nodeContext" : "new FailOnStartCommandProcessor(\"some crazy unexpected exception\")",
    "nodeType" : "ClassInstanceCreation",
    "nodePosition" : {
      "charLength" : 66,
      "startLineNumber" : 47,
      "startColumnNumber" : 8,
      "endLineNumber" : 47,
      "endColumnNumber" : 74
    },
    "astNodeNumber" : 4,
    "astHeight" : 3,
    "parentDataList" : [ {
      "locationInParent" : "ChildProperty[org.eclipse.jdt.core.dom.ExpressionStatement,expression]",
      "nodePosition" : {
        "charLength" : 104,
        "startLineNumber" : 46,
        "startColumnNumber" : 4,
        "endLineNumber" : 47,
        "endColumnNumber" : 74
      },
      "nodeContext" : "failOnStartCommandProcessor=new FailOnStartCommandProcessor(\"some crazy unexpected exception\")",
      "nodeType" : "Assignment",
      "astNodeNumber" : 6,
      "astHeight" : 4
    }, {
      "locationInParent" : "ChildListProperty[org.eclipse.jdt.core.dom.Block,statements]",
      "nodePosition" : {
        "charLength" : 105,
        "startLineNumber" : 46,
        "startColumnNumber" : 4,
        "endLineNumber" : 47,
        "endColumnNumber" : 75
      },
      "nodeContext" : "failOnStartCommandProcessor=new FailOnStartCommandProcessor(\"some crazy unexpected exception\");\n",
      "nodeType" : "ExpressionStatement",
      "astNodeNumber" : 7,
      "astHeight" : 5
    }, {
      "locationInParent" : "ChildProperty[org.eclipse.jdt.core.dom.MethodDeclaration,body]",
      "nodePosition" : {
        "charLength" : 589,
        "startLineNumber" : 44,
        "startColumnNumber" : 51,
        "endLineNumber" : 57,
        "endColumnNumber" : 3
      },
      "nodeContext" : "{\n  FailOnStartCommandProcessor failOnStartCommandProcessor;\n  failOnStartCommandProcessor=new FailOnStartCommandProcessor(\"some crazy unexpected exception\");\n  try {\n    new DefaultSelenium(failOnStartCommandProcessor).start();\n    fail(\"Did not catch RuntimeException as expected\");\n  }\n catch (  RuntimeException expected) {\n    assertTrue(-1 != expected.getMessage().indexOf(\"Could not start Selenium session: \"));\n    assertTrue(-1 != expected.getMessage().indexOf(\"some crazy unexpected exception\"));\n  }\n}\n",
      "nodeType" : "Block",
      "astNodeNumber" : 56,
      "astHeight" : 10
    }, {
      "locationInParent" : "ChildListProperty[org.eclipse.jdt.core.dom.TypeDeclaration,bodyDeclarations]",
      "nodePosition" : {
        "charLength" : 646,
        "startLineNumber" : 43,
        "startColumnNumber" : 2,
        "endLineNumber" : 57,
        "endColumnNumber" : 3
      },
      "nodeContext" : "@Test public void testShouldLeaveOtherExceptionAlone(){\n  FailOnStartCommandProcessor failOnStartCommandProcessor;\n  failOnStartCommandProcessor=new FailOnStartCommandProcessor(\"some crazy unexpected exception\");\n  try {\n    new DefaultSelenium(failOnStartCommandProcessor).start();\n    fail(\"Did not catch RuntimeException as expected\");\n  }\n catch (  RuntimeException expected) {\n    assertTrue(-1 != expected.getMessage().indexOf(\"Could not start Selenium session: \"));\n    assertTrue(-1 != expected.getMessage().indexOf(\"some crazy unexpected exception\"));\n  }\n}\n",
      "nodeType" : "MethodDeclaration",
      "astNodeNumber" : 62,
      "astHeight" : 11
    } ],
    "currentLineData" : {
      "locationInParent" : "ChildProperty[org.eclipse.jdt.core.dom.ExpressionStatement,expression]",
      "nodePosition" : {
        "charLength" : 104,
        "startLineNumber" : 46,
        "startColumnNumber" : 4,
        "endLineNumber" : 47,
        "endColumnNumber" : 74
      },
      "nodeContext" : "failOnStartCommandProcessor=new FailOnStartCommandProcessor(\"some crazy unexpected exception\")",
      "nodeType" : "Assignment",
      "astNodeNumber" : 6,
      "astHeight" : 4
    },
    "tokenLength" : 5,
    "type" : "com.thoughtworks.selenium.DefaultSeleniumStartErrorHandlingTest.FailOnStartCommandProcessor"
  } ],
  "positionList" : [ {
    "charLength" : 66,
    "startLineNumber" : 47,
    "startColumnNumber" : 8,
    "endLineNumber" : 47,
    "endColumnNumber" : 74
  } ],
  "layoutRelationDataList" : [ ]
}